frontend/src/scenes/authentication/twoFactorLogic.ts

```diff
@@ -153,19 +166,17 @@ export const twoFactorLogic = kea<twoFactorLogicType>([
         generateBackupCodesSuccess: () => {
             lemonToast.success('Backup codes generated successfully')
         },
-        toggleTwoFactorSetupModal: ({ open }) => {
-            if (!open) {
-                // Clear the form when closing the modal
-                actions.resetToken()
-            }
-        },
-        startSetup: async () => {
-            await api.get('api/users/@me/two_factor_start_setup/')
+        closeTwoFactorSetupModal: () => {
+            // Clear the form when closing the modal
+            actions.resetToken()
         },
     })),
 
-    afterMount(({ actions }) => {
-        actions.startSetup()
+    afterMount(({ actions, values }) => {
         actions.loadStatus()
+
+        if (values.user && values.user.organization?.enforce_2fa && !values.user.is_2fa_enabled) {
+            actions.openTwoFactorSetupModal(true)
```

Comment:
I feel like this is confusing because the modal could be already open (if it's forced open), yet we're saying here to open the modal. but instead of it opening the modal, what's actually happening is this side-effect where startSetup is called.

I think it technically works but it's confusing and took me a bit to understand the code.

Reply:
Curious - what is confusing about it? Do you have thoughts on a different way it could be done?